.github/workflows/clickhouse_ci.yml

```diff
@@ -61,18 +52,8 @@ jobs:
         run: pytest tests/integration_tests
 
       - name: Run ClickHouse Container (HEAD)
-        run: docker run
-          -d
-          -p 8123:8123
-          --name clickhouse_head
-          -v /var/lib/clickhouse
-          -v ${{ github.workspace }}/tests/integration_tests/test_config.xml:/etc/clickhouse-server/conf.d/test_config.xml
-          --ulimit nofile=262144:262144
-          clickhouse/clickhouse-server:head
+        run: CLICKHOUSE_VERSION=head docker compose up -d clickhouse
       - name: Run HEAD tests
         run: pytest tests/integration_tests
-      - name: stop head container
-        run: docker stop clickhouse_head
       - name: remove head container
-        run: docker remove clickhouse_head
-
+        run: docker compose down -v
```

Comment:
IMO, this needs splitting + a proper docker-compose action (like in the `on_push.yml`).